### PR TITLE
store_in_config() of backup parameter

### PR DESCRIPTION
**Pull request by mstormi**
and multiple cosmetics

Fixes: #1400

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

**Comment by ecdye** [comment]
@mstormi Please fix all of the testing issues before I review. Also please check the shellcheck bug that has occured over the last few days.

**Comment by mstormi** [comment]
Of course. It was late yesterday and I wanted to finish it by all means then forgot about that part bc I was so tired - sorry.

**Comment by ecdye** [comment]
> Of course. It was late yesterday and I wanted to finish it by all means then forgot about that part bc I was so tired - sorry.

You're fine, it happens to all of us, we're human.

**Comment by mstormi** [comment]
and today found out the whole parsing didn't work anyway :(

So here's a rewrite for you to check.

**Comment by mstormi** [comment]
Please give it an extra check w.r.t. adding/changing parameters in openhabian.conf please
(easiest probably is to replace openhabian.bash on your box with this one here, insert an 'exit 0' after update_openhabian_conf() in openhabian-setup.sh).
Add "new" fake parameters to openhabian.conf.dist, remove old ones, add fake params to /etc/openhabian.conf, add
a parameter to both with different values etc.
I don't want to miss anything that messes up someone's openhabian.conf.

File: functions/openhabian.bash [hunk 1]
```diff
@@ -63,38 +63,49 @@ openhabian_console_check() {
...
-    if [[ $line =~ ^# ]] || [[ -z "$line" ]]; then  # if line is a comment or empty
-      echo "$line"
-    else                                            # not fully sure if anything else is a proper param setting but let's assume this
-      param=$(echo "$line" | cut -d'=' -f1)         # get parameter name first
-      if [[ -z ${!param+x} ]]; then                 # if $param is set (if so it is because it was sourced on start)
-        echo "$line"                                # take the param from the default config by printing the default .conf's full line
-      elif [[ ${!param} == *" "* ]]; then
-      	echo "$param=\"${!param}\""                 # else parameter is set i.e. a line setting it is present in $config so use its value
+    if [[ $line =~ ^[a-zA-Z] ]]; then         # if line is a comment or empty
+      param=$(echo "$line" | cut -d'=' -f1)   # get parameter name first
```
Comment (ecdye) [inline functions/openhabian.bash:96]: Why are you changing this, if the line is a comment should we not just copy it over right away?

File: functions/packages.bash [hunk 2]
```diff
@@ -133,6 +133,7 @@ exim_setup() {
     if ! relayuser="$(whiptail --title "Enter your public service mail user" --inputbox "\\nEnter your mail username you use with the public service to relay all outgoing mail to $smarthost" 10 80 "$relayuser" 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
     if ! relaypass="$(whiptail --title "Enter your public service mail password" --passwordbox "\\nEnter the password to use for mailuser ${relayuser} to relay mail via ${smarthost}" 9 80 "$relaypass" 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
     if ! adminmail="$(whiptail --title "Enter your administration user's mail address" --inputbox "\\nEnter the address to send system reports to" 9 80 "$adminmail" 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
+    for i in smarthost smartport relayuser relaypass adminmail; do store_in_conf $i; done
```
Comment (ecdye) [inline functions/packages.bash:136]: Where is `store_in_conf` defined? I don't see it and don't recall it from prior code.

**Review by ecdye — CHANGES_REQUESTED** [review]
Have not tested yet but here are some preliminary thoughts.

File: functions/backup.bash [hunk 3]
```diff
@@ -263,79 +267,60 @@ create_amanda_config() {
...
+  local successText="Setup was successful.\\n\\nAmanda backup tool is now taking backups around 01:00. For further readings, start at http://wiki.zmanda.com/index.php/User_documentation."
 
   echo -n "$(timestamp) [openHABian] Beginning setup of the Amanda backup system... "
   if (whiptail --title "Amanda backup installation" --yes-button "Continue" --no-button "Cancel" --defaultno --yesno "$queryText" 24 80); then echo "OK"; else echo "CANCELED"; return 0; fi
 
-  if ! dpkg -s 'exim4' &> /dev/null; then
```
[the remark this answers is not in the document]
Comment (mstormi) [reply functions/backup.bash:297]: to only ask for admin mail address once (in exim setup) no matter if Amanda is setup or not

File: functions/config.bash [hunk 4]
```diff
@@ -58,10 +58,10 @@ update_config_java() {
       else
         echo "Zulu OpenJDK 64-bit: this option does not currently work on your platform. Defaulting to Java Zulu 8 32-bit installation."
       fi
-      if ! cond_redirect sed -i -e 's|^java_opt.*$|java_opt=Zulu8-32|' "$CONFIGFILE"; then return 1; fi
+      if ! cond_redirect sed -i -e 's|^java_opt.*$|java_opt=Zulu8-32|' "$configFile"; then return 1; fi
```
[the remark this answers is not in the document]
Comment (mstormi) [reply functions/config.bash:61]: done

File: functions/openhabian.bash (hunk 1 above)
Comment (mstormi) [reply functions/openhabian.bash:96]: I fully redid the function. The comment is bad - changed that.

File: functions/packages.bash (hunk 2 above)
Comment (mstormi) [reply functions/packages.bash:136]: openhabian@devpi:~/openhabian $ grep -n store_i */*
functions/openhabian.bash:73:##    store_in_conf(String parameter)
functions/openhabian.bash:75:store_in_conf() {

**Comment by ecdye** [comment]
Rebase on main and then I'll finish reviewing unless you think we should put this in OH2

**Comment by mstormi** [comment]
rebased on main but I don't think I can change this PR so created #1427

**Pull request closed without merging**